impl/src/ref_conv.rs

```diff
@@ -0,0 +1,312 @@
+use crate::utils::{add_where_clauses_for_new_ident, Either};
+use proc_macro2::{Ident, TokenStream};
+use quote::quote;
+use syn::{
+    parse::{Parse, ParseStream, Result},
+    spanned::Spanned,
+    DeriveInput, Field, Fields, Index, Token,
+};
+
+pub fn expand(
+    input: &DeriveInput,
+    trait_ident: &Ident,
+    method_ident: &Ident,
+    conv_type: &Ident,
+    mutability: Option<&Token![mut]>,
+) -> syn::Result<TokenStream> {
+    let trait_path = quote! { ::derive_more::#trait_ident };
+
+    let field_args = extract_field_args(input, trait_ident, method_ident)?;
+    let (impl_generics, ty_generics, where_clause) = input.generics.split_for_impl();
+    let input_type = &input.ident;
+
+    let sub_items: Vec<_> = field_args
+        .into_iter()
+        .map(
+            |FieldArgs {
+                 forward,
+                 field,
+                 ident,
+             }| {
+                let member = quote! { self.#ident };
+                let field_type = &field.ty;
+                if forward {
+                    let trait_path = quote! { #trait_path<#conv_type> };
+                    let type_where_clauses = quote! {
+                        where #field_type: #trait_path
+                    };
+                    let new_generics = add_where_clauses_for_new_ident(
```

Comment:
@MegaBluejay I would like to get rid of using this old function from `utils`, so we can rip old stuff totally, in the future. It also contains some logic (like adding `Copy` markers) which is totally irrelevant for these tratis.

Reply:
I kept the more basic functions (`add_extra_where_clauses`, `add_extra_generic_type_param`)

They seem general enough to be keep using everywhere